### ISSUE TITLE
"Ecosystem" and "Case Studies" are easily lost

### ISSUE THREAD

**Issue by bjnath** [numpy/numpy.org#238]
Ecosystem and Case Studies are strikingly original and attractive content --  and I'm concerned that because they're below the fold users will never see them.

I have suggestions to tighten the top of the page to pull at least one of them above the fold.

1. It's clear to readers that the boxes at the top are key features; the Key Features headline eats up space without adding content.

2. It's not crucial that the supporting text in those boxes always be displayed. The boxes can be turned into links, or if we're fancy can expand into boxes when clicked.

3. Finally, `Try NumPy` is a tour-de-force -- but it's competing with other content for visibility, and we have to ask the hard question of how much we benefit from having it on page one. It can't show NumPy's single most important feature -- speed;  though we've worked hard to dress it up, it's not graphically powerful; and it has no payoff unless the user stops reading to play with it, and we're not talking about pressing a button and watching. On a learning page -- it's killer. It's a beast. It needs the right home. On the home page, it can't add as much as Ecosystem and Case Studies do.

**Comment by joelachance** [comment]
@bjnath, I'll add some back story as to why the home page is in its current state. Originally, we were planning on having the interactive shell above the fold. However, Thebelab had more latency than we would have liked, as well as missing some functionality to really offer a clean experience. There's quite a few tickets and comments around this one that I won't go into here, but in short, we decided to move it lower on the home page.

With the shell lower, that bumped up key features, and as of very recently, news. We roughly styled key features after PyTorch's website section by the same name, and I personally wouldn't throw myself in front of a bus for what we've (I've) done there-- it's not my favorite, and I'll agree it feels it could use some additional work & love.

So, to comment on your points:
1 & 2. I'm a bit indifferent here, but I think the spirit of your second point might be to offer the thought that the UX could stand some improvement, in which 💯 . I think we're all open to suggestions (of which, I thought your suggestions were good ones!)

3.
> we have to ask the hard question of how much we benefit from having it on page one.

This seems like a fair question-- what does the user gain with an interactive shell on the homepage? Depends on the user. I don't think this benefits _all_ users, but I do think this is helpful for people who have no idea what NumPy is. We could say the same for the EcoSystem section-- a new user probably isn't going to dissect that section on their first visit.

> It can't show NumPy's single most important feature -- speed

I think this is a good point, although I don't think any in-browser solution can probably do this.

> it has no payoff unless the user stops reading to play with it

Can't this be said about everything on a website?

Overall, excellent points IMO.  I don't think we've necessarily given all of them the thought they may deserve, however, I do think that websites are tricky, and you want something for everyone. I don't think its in our best interest to put EcoSystem at the top-- the content is great, but it's not easy to read (there is so much content) and just isn't that welcoming for the casual numpy.org surfer. On the flip side, the shell probably isn't for the seasoned NumPy user-- they're going to headed straight for the docs, which we've got up top. I think the main benefit here is for others to try NumPy without having to do a `pip install`, seeing how intuitive the API is, and giving the website just a splash of fun v. walls of text. I'm not against moving it, but I'm not convinced `/learn` is the better place for it.

I'll note that @InessaPawson has done a lot of work to arrange the homepage (along with @rgommers ), so she likely will have some valuable insight here.

Sry for the crazy-long post, but thought it would be helpful. @bjnath, thank you for your suggestions, curious everyone else's thoughts!

**Comment by bjnath** [comment]
> Sry for the crazy-long post,

@joelachance, I much appreciate your comments!

Ecosystem should not be at the top -- totally agree. I had the thought but dropped it for the reasons you give -- and because it would just be bafflling. But our role in that immense hierarchy gives us a bragging right that no other site on the internet is likely to match. So I'm hoping to get it over the fold.

>  I don't think any in-browser solution can probably do this.

Oh, I agree! Here's my thinking. You can view a home page as having two purposes, the second of which is to show off and the first of which is to get people to another page.  We have three show-off sections, plus the shell, and it stands out because it's better for learning than boasting. We do have a section dedicated to learning, and people head there when they're feeling ambitious, and those are the people who will be delighted to find a see-for-yourself shell. 

> it has no payoff unless the user stops reading to play with it
>    Can't this be said about everything on a website?

If you could press a button and see the shell do tricks, that's to die for. The commitment needed to extract pleasure from the shell IRL lies several quanta beyond. You have to wait for it to start, and once it does, it's easy to get an error **that we're not prepared to help with**. Please understand I favor the shell, see the value of it, and would never keep it out. It's just a question of time and place. Ecosystem may be a lot to read, but it's got tabs to scroll through (i.e., pushbuttons) and each has pretty pictures. 

>  @InessaPawson has done a lot of work to arrange the homepage  (along with @rgommers )

Like you, eager to hear their feedback.

Thanks again for sharing the history and your thoughts.

**Comment by joelachance** [comment]
> So I'm hoping to get it over the fold.

I'll reiterate I think this content is just too dense for above the fold. Maybe there's other content that's easier to ingest that can still allow some humble brags?

> If you could press a button and see the shell do tricks, that's to die for. The commitment needed to extract pleasure from the shell IRL lies several quanta beyond.

I enjoyed that, very good :) Of course enjoyment is pretty subjective here, but I personally find myself playing with this stuff when I go to a site because it's more fun than reading a lot of text (haskell.org). Maybe I'm not the demographic around these parts, but I'd like to think you can put it in the 'show-off' column and not only the 'learning' column.
There's works better than ours, however, I've been working on Thebelab to improve some of the performance. @skopatz also has been working on https://github.com/regro/runthis-client which might be a good solution in the future as well.

**Comment by bjnath** [comment]
> I'll reiterate I think this content is just too dense for above the fold.

Sorry, I misread you.  The word "stark" comes up in only two contexts, and this is the other one. We have a stark difference of opinion.

One issue to consider with a page-one shell is handholding. Newbies and error messages go together. A learning page is equipped to walk users through their potential messages because it's a learning page. Page one is like a working parent; the kid's gotta manage on his own.

**Comment by rgommers** [comment]
> I'll reiterate I think this content is just too dense for above the fold.

I do still need to edit down the text a little, and change to a single graphic in the viz section - there's a bit too much.

**Comment by bjnath** [comment]
@rgommers as noted in issue #235 I have some concerns about wording in the Ecosystem section.

If you wish I can submit a PR for viz that has fewer words and also addresses #235.

**Comment by rgommers** [comment]
> If you wish I can submit a PR for viz that has fewer words and also addresses #235.

That would be great. I'll reply on gh-235

**Comment by InessaPawson** [comment]
I’ve covered extensively the motives behind several decisions in question  in this issue  https://github.com/numpy/numpy.org/issues/234#issuecomment-631889924

**Re: The Shell** As @joelachance mentioned, originally the shell was meant to be the main attraction of the homepage. (Please note, we don’t have a graphic designer on this project. Therefore we had to resort to other means of design entertainment. Not necessarily a bad thing. During my studies at SAIC designing with restrictions was encouraged, if not glorified.) When it became clear that the shell has certain shortcomings, I suggested changing the order of the content blocks to shift the focus away from it. I do think the shell belongs on the homepage given the page's major target audience. I’m also optimistic that the shell performance will get better. 

**Re: “Learn”** It is meant to be a curated collection of NumPy related educational resources. Giving the popularity and importance of the library, there are plenty of NumPy tutorials circulating on the World Wide Web. It is very helpful for NumPy users, especially beginners, to be able to find easily high-quality educational materials. I don’t think the shell belongs to the “Learn” page.

**Above the fold** If we wish to get more "bragging content" above the fold, “Case studies” would serve that purpose very well. However, the page's UI would suffer as we have too many boxes going on then.

**Comment by shaloo** [comment]
Folks, some of the recent edits on main page are really nice! Thanks for bringing those into effect.  That said, I would really appreciate if instead of throwing away original content we add a 'Learn More' button and allow those 'uninitiated to NumPy' to have a chance to grasp the full intent of what is being mentioned.   To NumPy experts, some content may look basic but remember, the website is not for experts primarily but to help those who want to 'begin' using NumPy.  Experts would typically go directly to dev docs or GitHub in most cases.

Good job @joelachance, @bjnath and everyone else behind the scenes that is helping make the landing page crisp and nice!

**Comment by bjnath** [comment]
Thank you, @shaloo!  Of course there'd **be** no landing page without @InessaPawson.

> throwing away original content

That's been nagging at me too.

> help those who want to 'begin' using NumPy

I agree -- we don't want the site to be unapproachable.

My feeling is that beginners come to the site because they're interested in visualization, data science, scientific domains, and so on. One of those ecosystem niches is what they're excited about, and NumPy is means to an end, something they need to know to get those things to work.

Viewed that way, a visualization user won't be hurt if we lack details on the visualization domain, and so on through the niches.

Is that plausible?

**Comment by rgommers** [comment]
Good point @shaloo. We also may want to be better at linking out to "learn more" places. E.g. https://pyviz.org/ is the best overview of visualization libraries.

**Comment by rgommers** [comment]
On the mailing list there was also a suggestion to move Ecosystem to the top: https://mail.python.org/pipermail/numpy-discussion/2020-May/080693.html. Just recording it here for now - let's postpone thinking about this till the next set of design improvements (see gh-266 for tracking issue for that).
